File: labs-algo/1/G.java
import java.util.ArrayList;
import java.util.Scanner;

/**
 * @author Saveliy Bakturin
 *
 * Don't write off, if you don't wanna be banned!
 */

public class G {
    public static void main(final String... args) {
        final Scanner in = new Scanner(System.in);
        final int n = in.nextInt(), m = in.nextInt();
        in.nextLine();
        final long[] a = new long[n];
        final long[] s = new long[n];
        long sum = 0;
        for (int i = 0; i < n; i++) {
            a[i] = in.nextLong();
            sum += a[i];
            s[i] = sum;
        }
        in.nextLine();
        final ArrayList<Long> answers = new ArrayList<>();
        for (int i = 0; i < m; i++) {
            String[] index = in.nextLine().split(" ");
            answers.add(s[Integer.parseInt(index[1]) - 1] - s[Integer.parseInt(index[0]) - 1] + a[Integer.parseInt(index[0]) - 1]);
        }
        in.close();
        for (final Long longer : answers) {
            System.out.println(longer);
        }
    }
}

File: algorithms/half/G/G.cpp
#include <cstdio>

/**
 * @author Saveliy Bakturin
 * <p>
 * Don't write off, if you don't wanna be banned!
 */

int main() {
	int n, m;
	scanf("%i %i", &n, &m);
	long long *a = new long long[n + 1]();
	for (int i = 1; i < n + 1; i++) {
		scanf("%lld", &a[i]);
		a[i] += a[i - 1];
	}
	for (int i = 0; i < m; i++) {
		int l, r;
		scanf("%i %i", &l, &r);
		printf("%lld\n", a[r] - a[l - 1]);
	}
}
